zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -166,7 +248,7 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             ),
         );
 
-        cb.condition(is_empty_code_hash.expr(), |cb| {
+        cb.condition(is_empty_code_hash.expr() + is_tx_invalid.expr(), |cb| {
```

Comment:
Better to use `or::expr` because both could be `1` (and then the constraints inside the condition could get multiplied by `2` which works for normal constraints but messes up lookups so better not do it).